commit 235dba3c798242ae1ca5bf420a0d47f2ace912c8 (refs/changes/67/5567/3)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2019-02-12T14:11:18-08:00 (8 months ago)
    
    TRITON-1215 ur.startup messages are useless to CNAPI
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/docs/cnapi-ha.md b/docs/cnapi-ha.md
index 1da2cb4..e954989 100644
--- a/docs/cnapi-ha.md
+++ b/docs/cnapi-ha.md
@@ -5,7 +5,7 @@
 -->
 
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright (c) 2019, Joyent, Inc.
 -->
 
 # CNAPI High Availability
@@ -28,9 +28,9 @@ faults.
 
 
 When any CNAPI instance starts up, it will connect to the `ur.cnapi` queue. It
-will then bind to this queue the routing keys, `ur.startup.#` and
-`ur.sysinfo.#`. When a compute node comes online, it will broadcast a message
-to a `ur.startup.#`.  Unsetup compute nodes will periodically emit `ur.sysinfo` messages to alert any
+will then bind to this queue the routing key `ur.sysinfo.#`. When a compute node
+comes online, it will broadcast a message to a `ur.sysinfo.#`.  Unsetup compute
+nodes will periodically emit `ur.sysinfo` messages after that to alert any
 listening CNAPI instances that the server exists, even though it may not yet
 have agents installed.
 
diff --git a/lib/app.js b/lib/app.js
index f526003..5d28bff 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -427,7 +427,6 @@ App.prototype.setupAmqpClient = function () {
     // Set up Ur client.
     self.log.debug('Ready to communicate with ur');
     self.ur = new Ur({ log: self.log });
-    self.ur.on('serverStartup', self.onSysinfoReceivedUr.bind(self));
     self.ur.on('serverSysinfo', self.onSysinfoReceivedUr.bind(self));
 };
 
diff --git a/lib/ur.js b/lib/ur.js
index c95b400..0a058e4 100644
--- a/lib/ur.js
+++ b/lib/ur.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -46,14 +46,11 @@ Ur.prototype.bindQueues = function () {
 
     self.log.info('attempting to open Ur queue');
     var q = self.connection.queue(self.queue, function (queue) {
-        queue.bind('ur.startup.#');
         queue.bind('ur.sysinfo.#');
 
         self.log.info('bound and listening on Ur queue');
         queue.subscribeJSON(function (message, headers, deliveryInfo) {
-            if (deliveryInfo.routingKey.split('.')[1] === 'startup') {
-                self.emit('serverStartup', message, deliveryInfo.routingKey);
-            } else if (deliveryInfo.routingKey.split('.')[1] === 'sysinfo') {
+            if (deliveryInfo.routingKey.split('.')[1] === 'sysinfo') {
                 self.emit('serverSysinfo', message, deliveryInfo.routingKey);
             }
         });
diff --git a/package.json b/package.json
index b6de70c..0aeb2dc 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.19.1",
+  "version": "1.19.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
